src/network/networkd-link.c

```diff
@@ -71,6 +71,20 @@
 #include "udev-util.h"
 #include "vrf.h"
 
+void link_required_operstate_for_online(Link *link, LinkOperationalStateRange *ret) {
```

Comment:
why not define this as `LinkOperationStateRange link_required_operstate_for_online(Link *link)`, i.e. return the result?

I mean, if it cannot fail, and currently returns void...